cmd/fanoutsidecar/main.go

```diff
@@ -108,9 +109,11 @@ func main() {
 	})
 	logger.Info("Fanout sidecar Listening...", zap.String("Address", s.Addr))
 	g.Go(s.ListenAndServe)
-	if err = g.Wait(); err != nil {
-		logger.Fatal("Either the HTTP server or the ConfigMap noticer failed.", zap.Error(err))
+	err = g.Wait()
+	if err != nil {
+		logger.Error("Either the HTTP server or the ConfigMap noticer failed.", zap.Error(err))
 	}
+	s.Shutdown(context.TODO())
```

Comment:
Optional: setting a timeout here ensures the process actually terminates eventually.
```go
// timeout shutdown after 10 seconds
ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
defer cancel()
s.Shutdown(ctx)
```

Reply:
Done.